src/objects/obstacles/pipes.ts

```diff
@@ -89,20 +109,61 @@ export class Pipes extends BaseObject {
 
     }
 
-    update(){
+    /**
+     * Update difficulty level based on elapsed time.
+     * @param currentTime Current game time in ms
+     */
+    private updateDifficulty(currentTime: number) {
+        const elapsedTime = currentTime - this.gameStartTime;
+        if(elapsedTime >= 2 * this.difficultyInterval){
+            this.difficultyLevel = 3;
+            this.pipeGapX = CONFIG.PIPE_GAP_X_Level_3;
```

Comment:
longer term, you don't really need constants, you could just -50 the gap every time difficulty goes up. or -50*level or something. Its usually easier to maintain than trying to have all the numbers as new constants

is fine for now though

Reply:
If you store the targetLevel time or something, instead of trying to do math for elapsed, you could do

if (currentTime > targetEndTime) {
  this.difficultyLevel++;
  this.pipeGapX -= 50;
  this.pipeGapY -= 50;
}

or something?